build.gradle.kts

```diff
@@ -0,0 +1,170 @@
+      storePassword = System.getenv("DEBUG_KEYSTORE_PASSWORD") ?: "debug0"
+      keyAlias = System.getenv("DEBUG_KEY_ALIAS") ?: "debug"
+      keyPassword = System.getenv("DEBUG_KEY_PASSWORD") ?: "debug0"
+    }
+
+    create("release") {
+      val ks = System.getenv("RELEASE_KEYSTORE")
+      if (ks != null) {
+        storeFile = file(ks)
+        storePassword = System.getenv("RELEASE_KEYSTORE_PASSWORD")
+        keyAlias = System.getenv("RELEASE_KEY_ALIAS")
+        keyPassword = System.getenv("RELEASE_KEY_PASSWORD")
+      }
+    }
+  }
+
+  buildTypes {
+    named("release") {
+      isMinifyEnabled = true
+      isShrinkResources = true
+      isDebuggable = false
+      proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"))
+      resValue("string", "app_name", "@string/app_name_release")
+      signingConfig = signingConfigs["release"]
+    }
+
+    named("debug") {
+      isMinifyEnabled = false
+      isShrinkResources = false
+      isDebuggable = true
+      applicationIdSuffix = ".debug"
+      resValue("string", "app_name", "@string/app_name_debug")
+      resValue("bool", "debug_logs", "true")
+      signingConfig = signingConfigs["debug"]
+    }
+  }
+
+  // Name outputs after the application ID.
+  android.applicationVariants.forEach { variant ->
+    variant.outputs.forEach {
+      it as BaseVariantOutputImpl
+      it.outputFileName = "${variant.applicationId}.apk"
+    }
+  }
+
+  compileOptions {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+  }
+}
+
+val buildKeyboardFont by tasks.registering(Exec::class) {
+  doFirst {
+    println("\nBuilding assets/special_font.ttf")
+    mkdir(layout.buildDirectory)
+  }
+  workingDir = projectDir.resolve("srcs/special_font")
+  val svgFiles = workingDir.listFiles()!!.filter {
+    it.isFile && it.name.endsWith(".svg")
+  }.toTypedArray()
+  val output = layout.buildDirectory.file("special_font.ttf")
+  commandLine("fontforge", "-lang=ff", "-script", "build.pe", output.get().asFile.absolutePath, *svgFiles)
+  doLast {
+    copy {
+      from(output)
+      into("assets")
+    }
+  }
+}
+
+val genEmojis by tasks.registering(Exec::class) {
+  doFirst { println("\nGenerating res/raw/emojis.txt") }
+  workingDir = projectDir
+  commandLine("python", "gen_emoji.py")
+}
+
+val genLayoutsList by tasks.registering(Exec::class) {
+  doFirst { println("\nGenerating res/values/layouts.xml") }
+  workingDir = projectDir
+  commandLine("python", "gen_layouts.py")
+}
+
+val checkKeyboardLayouts by tasks.registering(Exec::class) {
+  doFirst { println("\nChecking layouts") }
+  workingDir = projectDir
+  commandLine("python", "check_layout.py")
+}
+
+val compileComposeSequences by tasks.registering(Exec::class) {
+  val out = "srcs/juloo.keyboard2/ComposeKeyData.java"
+  doFirst { println("\nGenerating $out") }
+  val sequences = projectDir.resolve("srcs/compose").listFiles {
```

Comment:
I get these errors and I'm not sure how to resolve them:

```
e: unexpected-keyboard/build.gradle.kts:137:54: Overload resolution ambiguity: 
public open fun listFiles(p0: FileFilter!): Array<(out) File!>! defined in java.io.File
public open fun listFiles(p0: FilenameFilter!): Array<(out) File!>! defined in java.io.File
e: unexpected-keyboard/build.gradle.kts:138:6: Unresolved reference: it
e: unexpected-keyboard/build.gradle.kts:138:34: Unresolved reference: it
e: unexpected-keyboard/build.gradle.kts:139:13: Unresolved reference: it
```

I'm using `8.14.3` and I can't upgrade easily.

Reply:
This is because 8.14.3 ships an outdated version of Kotlin that seemingly doesn't handle this case properly.
Since I have now downgraded Gradle, this has been fixed by providing more type information.